### PR TITLE
feat(aws): Adding property externalId to AWS account config for assuming roles.

### PR DESCRIPTION
**Pull request by ivanphdz**
Enabling the use of the externalId parameter when assuming a role prevents the "confused deputy" problem.

https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html#external-id-purpose

### CODE REVIEW

**Comment by fieldju** [comment]
@ivanphdz can you update 👇  to include externalId with a comment

https://github.com/spinnaker/clouddriver/blob/master/clouddriver-web/config/clouddriver.yml#L318-L330

**Comment by fieldju** [comment]
After this PR is merged, we are working on making sure Halyard and Kleat get patched to support this feature as well.

**Review by fieldju — APPROVED** [review]

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/security/NetflixSTSAssumeRoleSessionCredentialsProvider.java [hunk 1]
```diff
@@ -17,33 +17,63 @@
...
       String roleSessionName,
-      String accountId) {
-    super(longLivedCredentialsProvider, roleArn, roleSessionName);
+      String accountId,
+      String externalId) {
     this.accountId = accountId;
 
+    delegate =
+        new STSAssumeRoleSessionCredentialsProvider.Builder(roleArn, roleSessionName)
+            .withExternalId(externalId)
```
Comment (kevinawoo) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/security/NetflixSTSAssumeRoleSessionCredentialsProvider.java:42]: does AWS SDK allow this to be nullable?

**Review by kevinawoo — APPROVED** [review]

**Comment by plumpy** [comment]
This is [preventing clouddriver from starting up on Jenkins](https://builds.spinnaker.io/job/Validate_BomMultiPlatform/5977/CLOUD_PLATFORM=gce/artifact/logs/service_logs/clouddriver.log):

```
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]: Caused by: com.amazonaws.SdkClientException: Unable to find a region via the region provider chain. Must provide an explicit region in the builder or setup environment to supply a region.
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.amazonaws.client.builder.AwsClientBuilder.setRegion(AwsClientBuilder.java:462) ~[aws-java-sdk-core-1.11.820.jar:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.amazonaws.client.builder.AwsClientBuilder.configureMutableProperties(AwsClientBuilder.java:424) ~[aws-java-sdk-core-1.11.820.jar:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.amazonaws.client.builder.AwsSyncClientBuilder.build(AwsSyncClientBuilder.java:46) ~[aws-java-sdk-core-1.11.820.jar:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.NetflixSTSAssumeRoleSessionCredentialsProvider.<init>(NetflixSTSAssumeRoleSessionCredentialsProvider.java:46) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.AssumeRoleAmazonCredentials.createSTSCredentialsProvider(AssumeRoleAmazonCredentials.java:60) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.NetflixAssumeRoleAmazonCredentials.<init>(NetflixAssumeRoleAmazonCredentials.java:161) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.NetflixAssumeRoleAmazonCredentials.<init>(NetflixAssumeRoleAmazonCredentials.java:92) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.config.CredentialsLoader$CopyConstructorTranslator.translate(CredentialsLoader.java:397) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.config.CredentialsLoader.load(CredentialsLoader.java:303) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.config.CredentialsLoader$load.call(Unknown Source) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47) ~[groovy-2.5.11.jar:2.5.11]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115) ~[groovy-2.5.11.jar:2.5.11]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127) ~[groovy-2.5.11.jar:2.5.11]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.DefaultAmazonAccountsSynchronizer.synchronize(DefaultAmazonAccountsSynchronizer.groovy:46) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.AmazonAccountsSynchronizer$synchronize.call(Unknown Source) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47) ~[groovy-2.5.11.jar:2.5.11]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115) ~[groovy-2.5.11.jar:2.5.11]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentialsInitializer.netflixAmazonCredentials(AmazonCredentialsInitializer.groovy:71) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentialsInitializer$$EnhancerBySpringCGLIB$$b7b878e3.CGLIB$netflixAmazonCredentials$8(<generated>) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentialsInitializer$$EnhancerBySpringCGLIB$$b7b878e3$$FastClassBySpringCGLIB$$1e171471.invoke(<generated>) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.2.9.RELEASE.jar:5.2.9.RELEASE]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331) ~[spring-context-5.2.9.RELEASE.jar:5.2.9.RELEASE]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentialsInitializer$$EnhancerBySpringCGLIB$$b7b878e3.netflixAmazonCredentials(<generated>) ~[clouddriver-aws-7.0.0-20201007110019.jar:7.0.0-20201007110019]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.2.9.RELEASE.jar:5.2.9.RELEASE]
Oct 07 16:39:33 jenkins-validate-bom-vmg-5977 clouddriver[24773]:         ... 84 common frames omitted
```

**Pull request merged**